Tribler/Main/vwxGUI/list.py

```diff
@@ -302,6 +305,77 @@ def downloadStarted(self, _):
         self.refresh()
 
 
+class CreditMiningSearchManager(BaseManager):
+    """
+    search manager for credit mining purpose
+    """
+    def __init__(self, llist):
+        BaseManager.__init__(self, llist)
+        self.boosting_manager = BoostingManager.get_instance()
+        self.library_manager = self.guiutility.library_manager
+
+    def refresh(self):
+        startWorker(self._on_data, self.get_torrents_list_boosting, uId=u"CreditMiningSearchManager_refresh",
+                    retryOnBusy=True, priority=GUI_PRI_DISPERSY)
+
+    def get_torrent_from_infohash(self, infohash):
+        torrent = self.boosting_manager.torrents.get(infohash, None)
+        if torrent:
+            ltorrent = LibraryTorrent('', infohash, name=torrent['name'], length=torrent['length'], category='',
+                                      status='', num_seeders=torrent['num_seeders'],
+                                      num_leechers=torrent['num_leechers'])
+            ltorrent.torrent_db = self.library_manager.torrent_db
+            ltorrent.channelcast_db = self.library_manager.channelcast_db
+
+            #touch channel instance
+            ltorrent.channel
```

Comment:
For line 331, it's been like that since I forked this from @whirm repo. What's this, Elric?